### PR TITLE
Make room for stencil data in host buffer

### PR DESCRIPTION
**Pull request by asuonpaa**
When copying a D24_UNORM_S8_UINT depth/stencil buffer to or from the host, the depth and stencil data must be copied separately. This change adds extra space to the end of the Amber host buffer for the stencil data.

### CODE REVIEW

File: src/vulkan/transfer_image.cc [hunk 1]
```diff
@@ -77,7 +77,12 @@ TransferImage::TransferImage(Device* device,
                              uint32_t base_mip_level,
                              uint32_t used_mip_levels,
                              uint32_t samples)
-    : Resource(device, x * y * z * format.SizeInBytes()),
+    : Resource(
+          device,
+          x * y * z *
+              (format.SizeInBytes() +
+               (format.GetFormatType() == FormatType::kD24_UNORM_S8_UINT ? 1
+                                                                         : 0))),
```
Comment (paulthomson) [inline src/vulkan/transfer_image.cc:85]: We should add a comment to explain why.

**Review by paulthomson — COMMENTED** [review]
Looks good! However, take a look at:

* `GraphicsPipeline::Clear`
* `GraphicsPipeline::Draw`

These both call `CopyBuffersToImages` and `TransferImagesToDevice`. Perhaps we could remove the code in these functions (and the functions that they call) that handles the depth stencil buffers, and instead handle them at a higher level (e.g. in `GraphicsPipeline::Draw`). Perhaps this would make it easier to create temporary buffers etc. and interleave the stencil data as needed.

File: src/vulkan/transfer_image.cc (hunk 1 above)
Comment (asuonpaa) [reply src/vulkan/transfer_image.cc]: done

**Comment by asuonpaa** [comment]
> Looks good! However, take a look at:
> 
> * `GraphicsPipeline::Clear`
> * `GraphicsPipeline::Draw`
> 
> These both call `CopyBuffersToImages` and `TransferImagesToDevice`. Perhaps we could remove the code in these functions (and the functions that they call) that handles the depth stencil buffers, and instead handle them at a higher level (e.g. in `GraphicsPipeline::Draw`). Perhaps this would make it easier to create temporary buffers etc. and interleave the stencil data as needed.

I don't see why we'd want to add code that prepares for interleaving unless we add the functionality with the same commit. Also wouldn't it just add redundancy if we moved the buffer handling higher where it would be required in multiple places?

**Review by paulthomson — APPROVED** [review]

**Pull request merged**